<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>566</width>
    <height>512</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>QString函数测试</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_11">
   <item>
    <layout class="QVBoxLayout" name="verticalLayout" stretch="1,1,1,1">
     <property name="spacing">
      <number>0</number>
     </property>
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>字符串截取</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_7">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_9">
          <property name="rightMargin">
           <number>4</number>
          </property>
          <item>
           <widget class="QPushButton" name="pushButton_QStr_front_back">
            <property name="text">
             <string>front &amp; back</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_QStr_left_right">
            <property name="text">
             <string>left &amp; right</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_10">
          <property name="spacing">
           <number>6</number>
          </property>
          <property name="rightMargin">
           <number>4</number>
          </property>
          <item>
           <widget class="QPushButton" name="pushButton_QStr_first_last">
            <property name="text">
             <string>first &amp; last</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_QStr_section">
            <property name="text">
             <string>section *</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="title">
        <string>存储相关</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_7">
          <property name="rightMargin">
           <number>4</number>
          </property>
          <item>
           <widget class="QPushButton" name="pushButton_QStr_isN_isEmpty">
            <property name="text">
             <string>isNull &amp; isEmpty</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_QStr_resize">
            <property name="text">
             <string>resize</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_8">
          <property name="spacing">
           <number>6</number>
          </property>
          <property name="rightMargin">
           <number>4</number>
          </property>
          <item>
           <widget class="QPushButton" name="pushButton_size_length">
            <property name="text">
             <string>size &amp; length</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_QStr_fill">
            <property name="text">
             <string>fill</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_4">
       <property name="title">
        <string>转换或修改</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="0,0">
          <property name="spacing">
           <number>6</number>
          </property>
          <property name="rightMargin">
           <number>4</number>
          </property>
          <item>
           <widget class="QPushButton" name="pushButton_QStr_trim_simplifiy">
            <property name="text">
             <string>trimmed &amp; simplified</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_QStr_insert">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>insert</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <property name="rightMargin">
           <number>4</number>
          </property>
          <item>
           <widget class="QPushButton" name="pushButton_QStr_remove">
            <property name="text">
             <string>remove</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_QStr_replace">
            <property name="text">
             <string>replace</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_3">
       <property name="title">
        <string>搜索和判断</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <property name="rightMargin">
           <number>4</number>
          </property>
          <item>
           <widget class="QPushButton" name="pushButton_QStr_indexof">
            <property name="text">
             <string>indexof &amp; lastIndexOf</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_QStr_contains">
            <property name="text">
             <string>contains</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <property name="rightMargin">
           <number>4</number>
          </property>
          <item>
           <widget class="QPushButton" name="pushButton_QStr_endsWith_startsWith">
            <property name="text">
             <string>endsWith &amp; startsWith</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_QStr_count">
            <property name="text">
             <string>count</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_3" stretch="1,3">
     <property name="spacing">
      <number>0</number>
     </property>
     <item>
      <widget class="QGroupBox" name="groupBox_5">
       <property name="title">
        <string>输入</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_8" stretch="0,0">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,4,0">
          <property name="spacing">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>4</number>
          </property>
          <item>
           <widget class="QLabel" name="label">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>str1</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="comboBox_QStr_str1">
            <item>
             <property name="text">
              <string>String 1,男，2013-03-12，北京，华北地区，中国</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>C://programm/Qt/QT/test.cpp/test2.cpp</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="1,3,1,3">
          <property name="spacing">
           <number>2</number>
          </property>
          <property name="rightMargin">
           <number>4</number>
          </property>
          <item>
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>str2</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="comboBox_QStr_str2">
            <item>
             <property name="text">
              <string>C://div/one/to///thre//fout/.cpp</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>String2 aaaaaa  find something.      so   .. .. .        </string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>参数</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spinBox">
            <property name="buttonSymbols">
             <enum>QAbstractSpinBox::PlusMinus</enum>
            </property>
            <property name="suffix">
             <string extracomment=" (&lt;100)"/>
            </property>
            <property name="maximum">
             <number>100</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_6">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="title">
        <string>结果显示</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_9">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <property name="spacing">
           <number>2</number>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>3</number>
          </property>
          <property name="bottomMargin">
           <number>1</number>
          </property>
          <item>
           <widget class="QPushButton" name="pushButton_QStr_clearTxt">
            <property name="text">
             <string>清空文本框</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPlainTextEdit" name="plainTextEdit">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
